include/RAJA/pattern/kernel/OffSet.hpp

```diff
@@ -0,0 +1,65 @@
+/*!
```

Comment:
These are the 'helpers' you mentioned above, correct??  I know why you did it this way, but the implementations are all the same. Have you thought about making one implementation that is templated on the "index" type?  For example, instead of ParamList<0, 1>, what about something like List<Param>(0, 1)? List is a bad name here, but we could come up with something better.  @ajkunen what do you think about this?

Reply:
I think we need to have the variants to identify from which tuple the data should come from, Segment tuple vs Param tuple.

Reply:
Fixed with latest iteration.